client/maps/maps-reducer.ts

```diff
@@ -0,0 +1,126 @@
+import { Immutable } from 'immer'
+import { MapInfoJson } from '../../common/maps'
+import { FetchError } from '../network/fetch-action-types'
+import { immerKeyedReducer } from '../reducers/keyed-reducer'
+
+export interface MapsState {
+  /** A map of map ID -> information about the map. */
+  byId: Map<string, MapInfoJson>
+  /** A map of the favorited map ID -> information about the favorited map. */
+  favoritedById: Map<string, MapInfoJson>
+  /** Total number of maps currently saved in a store. */
+  total: number
+  /** A flag indicating whether we're currently requesting maps. */
+  isRequesting: boolean
+  /** A set of map IDs that the user is currently (un)favoriting. */
+  favoriteStatusRequests: Set<string>
+  /** A potential error of our request to get maps. */
+  lastError?: FetchError
+}
+
+const DEFAULT_STATE: Immutable<MapsState> = {
+  byId: new Map(),
+  favoritedById: new Map(),
+  total: -1,
```

Comment:
would probably prefer `0` over `-1` for stuff like this, it functions better if we end up with that value at some point (e.g. `if (!state.total)` will function properly with `0`, but not `-1`)